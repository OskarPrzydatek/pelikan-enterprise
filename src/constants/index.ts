export enum Icons {
  ADD,
  ARROW,
  CHEVRON,
  CLOSE,
  DELETE,
  EDIT,
  HAMBURGER,
}

export enum Slugs {
  ATTRACTION_OVERVIEW = 'attraction-overview',
  CREATE_ATTRACTION = 'create-attraction',
  EDIT_ATTRACTION = 'edit-attraction',

  HASHTAG_OVERVIEW = 'hashtag-overview',
  CREATE_HASHTAG = 'create-hashtag',
  EDIT_HASHTAG = 'edit-hashtag',

  HOTEL_OVERVIEW = 'hotel-overview',
  CREATE_HOTEL = 'create-hotel',
  EDIT_HOTEL = 'edit-hotel',

  OFFER_OVERVIEW = 'offer-overview',
  CREATE_OFFER = 'create-offer',
  EDIT_OFFER = 'edit-offer',

  TRANSPORT_OVERVIEW = 'transport-overview',
  CREATE_TRANSPORT = 'create-transport',
  EDIT_TRANSPORT = 'edit-transport',

  USERS_OVERVIEW = 'users-overview',
  CREATE_USER = 'create-user',
  EDIT_USER = 'edit-user',

  LOGIN = 'LOGIN',

  ID = ':id',
  PAGE = ':page',
  NOT_FOUND = '*',
}

export enum Endpoints {
  ATTRACTIONS_LIST = 'attractions/list',
  ATTRACTION_BY_ID = 'attractions/get',
  CREATE_ATTRACTION = 'attractions/add',
  UPDATE_ATTRACTION = 'attractions/update',
  DELETE_ATTRACTION = 'attractions/delete',

  HASHTAGS_LIST = 'hashtags/list',
  HASHTAG_BY_ID = 'hashtags/get',
  CREATE_HASHTAG = 'hashtags/add',
  UPDATE_HASHTAG = 'hashtags/update',
  DELETE_HASHTAG = 'hashtags/delete',

  HOTELS_LIST = 'hotels/list',
  HOTEL_BY_ID = 'hotels/get',
  CREATE_HOTEL = 'hotels/add',
  UPDATE_HOTEL = 'hotels/update',
  DELETE_HOTEL = 'hotels/delete',

  OFFERS_LIST = 'offers/list',
  OFFER_BY_ID = 'offers/get',
  CREATE_OFFER = 'offers/add',
  UPDATE_OFFER = 'offers/update',
  DELETE_OFFER = 'offers/delete',

  TRANSPORTS_LIST = 'transports/list',
  TRANSPORT_BY_ID = 'transports/get',
  CREATE_TRANSPORT = 'transports/add',
  UPDATE_TRANSPORT = 'transports/update',
  DELETE_TRANSPORT = 'transports/delete',

  USERS_LIST = 'users/list',
  USER_BY_ID = 'users/get',
  CREATE_USER = 'users/add',
  UPDATE_USER = 'users/update',
  DELETE_USER = 'users/delete',
}

export const transportTypeOptions = [
  { value: 'PLANE', label: 'Samolot' },
  { value: 'BUS', label: 'Autobus' },
  { value: 'SHIP', label: 'Statek' },
];

export const userRoleType = [
  { value: 'ADMIN', label: 'Administrator' },
  { value: 'WORKER', label: 'Pracownik' },
];
