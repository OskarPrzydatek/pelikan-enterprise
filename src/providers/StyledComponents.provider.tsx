import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
