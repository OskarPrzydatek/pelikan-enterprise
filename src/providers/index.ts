export { StyledComponentsProvider } from './StyledComponents.provider';
