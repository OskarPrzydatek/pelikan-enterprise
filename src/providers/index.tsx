import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

interface ITestingPageProvider {
  element: JSX.Element;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};

export const TestingPageProvider: React.FC<ITestingPageProvider> = ({
  element,
}: ITestingPageProvider) => {
  return (
    <StyledComponentsProvider>
      <MemoryRouter initialEntries={['/page1']}>
        <Routes>
          <Route element={element} path="/:page" />
        </Routes>
      </MemoryRouter>
    </StyledComponentsProvider>
  );
};
