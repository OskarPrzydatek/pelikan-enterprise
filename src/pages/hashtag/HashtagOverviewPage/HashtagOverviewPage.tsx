import { useNavigate, useParams } from 'react-router-dom';

import { HashtagOverviewTemplate } from '~/components/templates';
import { Slugs } from '~/constants';

export const HashtagOverviewPage: React.FC = () => {
  const { page } = useParams();
  const navigate = useNavigate();

  const onClickNavigateToCreateHashtag = () =>
    navigate(`/${Slugs.CREATE_HASHTAG}`);

  return (
    <HashtagOverviewTemplate
      page={page}
      onClickNavigate={onClickNavigateToCreateHashtag}
    />
  );
};
