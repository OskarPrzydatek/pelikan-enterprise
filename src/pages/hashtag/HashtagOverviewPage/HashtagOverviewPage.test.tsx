import { fireEvent, render, screen } from '@testing-library/react';
import useSWR from 'swr';
import { Mock, describe, test, vi } from 'vitest';

import { Slugs } from '~/constants';
import { hashtagDataMock } from '~/mocks';
import { TestingPageProvider } from '~/providers';

import { HashtagOverviewPage } from './HashtagOverviewPage';

const MockHashtagOverviewPage: React.FC = () => (
  <TestingPageProvider element={<HashtagOverviewPage />} />
);

const mockNavigate = vi.fn();

vi.mock('swr', async () => {
  const actual = await vi.importActual<object>('swr');

  return {
    ...actual,
  };
});

vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual<{}>('react-router-dom');
  return {
    ...actual,
    useNavigate: () => mockNavigate,
  };
});

describe('HashtagOverviewPage', () => {
  test('page snapshot', () => {
    const view = render(<MockHashtagOverviewPage />);
    expect(view).toMatchSnapshot();
  });
});
