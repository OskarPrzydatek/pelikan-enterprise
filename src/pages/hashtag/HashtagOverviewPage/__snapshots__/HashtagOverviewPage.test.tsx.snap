// Vitest Snapshot v1

exports[`HashtagOverviewPage > page snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-ezOQGI deUqHb"
      >
        <div
          class="sc-cabOPr isyJVs"
        >
          <h2
            class="sc-eDvSVe dcXtK"
          >
            Hashtagi
          </h2>
          <ul
            class="sc-iTFTee bMSplb"
          >
            <li
              class="sc-eDWCr jtwhuZ"
              data-testid="hashtag-overview-list-item-1"
            >
              <p
                class="sc-pyfCe bsnSHf"
              >
                Travel
              </p>
              <div
                class="sc-bqWxrE kyFLpy"
              >
                <button
                  class="sc-jSUZER fgGcDJ"
                  data-testid="hashtag-overview-edit-1"
                >
                  <svg
                    data-testid="edit-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="22.5461"
                      stroke="#013059"
                      stroke-width="3"
                      width="21.0433"
                      x="2.05469"
                      y="7.50781"
                    />
                    <rect
                      fill="white"
                      height="12.0217"
                      width="12.0217"
                      x="15.5781"
                      y="2.99219"
                    />
                    <path
                      d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                      fill="#013059"
                    />
                    <rect
                      fill="#013059"
                      height="2.59198"
                      transform="rotate(-135 29.7031 5.03906)"
                      width="4.53906"
                      x="29.7031"
                      y="5.03906"
                    />
                  </svg>
                </button>
                <button
                  class="sc-jSUZER fgGcDJ"
                  data-testid="hashtag-overview-delete-1"
                >
                  <svg
                    data-testid="delete-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 22 33"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                      stroke="#F23557"
                      stroke-width="3"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="2"
                      x1="0.226562"
                      x2="21.4318"
                      y1="3.72656"
                      y2="3.72656"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="3"
                      x1="7.29688"
                      x2="14.3653"
                      y1="1.82031"
                      y2="1.82031"
                    />
                  </svg>
                </button>
              </div>
            </li>
            <li
              class="sc-eDWCr jtwhuZ"
              data-testid="hashtag-overview-list-item-2"
            >
              <p
                class="sc-pyfCe bsnSHf"
              >
                Adventure
              </p>
              <div
                class="sc-bqWxrE kyFLpy"
              >
                <button
                  class="sc-jSUZER fgGcDJ"
                  data-testid="hashtag-overview-edit-2"
                >
                  <svg
                    data-testid="edit-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="22.5461"
                      stroke="#013059"
                      stroke-width="3"
                      width="21.0433"
                      x="2.05469"
                      y="7.50781"
                    />
                    <rect
                      fill="white"
                      height="12.0217"
                      width="12.0217"
                      x="15.5781"
                      y="2.99219"
                    />
                    <path
                      d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                      fill="#013059"
                    />
                    <rect
                      fill="#013059"
                      height="2.59198"
                      transform="rotate(-135 29.7031 5.03906)"
                      width="4.53906"
                      x="29.7031"
                      y="5.03906"
                    />
                  </svg>
                </button>
                <button
                  class="sc-jSUZER fgGcDJ"
                  data-testid="hashtag-overview-delete-2"
                >
                  <svg
                    data-testid="delete-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 22 33"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                      stroke="#F23557"
                      stroke-width="3"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="2"
                      x1="0.226562"
                      x2="21.4318"
                      y1="3.72656"
                      y2="3.72656"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="3"
                      x1="7.29688"
                      x2="14.3653"
                      y1="1.82031"
                      y2="1.82031"
                    />
                  </svg>
                </button>
              </div>
            </li>
            <li
              class="sc-eDWCr jtwhuZ"
              data-testid="hashtag-overview-list-item-3"
            >
              <p
                class="sc-pyfCe bsnSHf"
              >
                Explore
              </p>
              <div
                class="sc-bqWxrE kyFLpy"
              >
                <button
                  class="sc-jSUZER fgGcDJ"
                  data-testid="hashtag-overview-edit-3"
                >
                  <svg
                    data-testid="edit-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="22.5461"
                      stroke="#013059"
                      stroke-width="3"
                      width="21.0433"
                      x="2.05469"
                      y="7.50781"
                    />
                    <rect
                      fill="white"
                      height="12.0217"
                      width="12.0217"
                      x="15.5781"
                      y="2.99219"
                    />
                    <path
                      d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                      fill="#013059"
                    />
                    <rect
                      fill="#013059"
                      height="2.59198"
                      transform="rotate(-135 29.7031 5.03906)"
                      width="4.53906"
                      x="29.7031"
                      y="5.03906"
                    />
                  </svg>
                </button>
                <button
                  class="sc-jSUZER fgGcDJ"
                  data-testid="hashtag-overview-delete-3"
                >
                  <svg
                    data-testid="delete-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 22 33"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                      stroke="#F23557"
                      stroke-width="3"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="2"
                      x1="0.226562"
                      x2="21.4318"
                      y1="3.72656"
                      y2="3.72656"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="3"
                      x1="7.29688"
                      x2="14.3653"
                      y1="1.82031"
                      y2="1.82031"
                    />
                  </svg>
                </button>
              </div>
            </li>
            <li
              class="sc-eDWCr jtwhuZ"
              data-testid="hashtag-overview-list-item-4"
            >
              <p
                class="sc-pyfCe bsnSHf"
              >
                Nature
              </p>
              <div
                class="sc-bqWxrE kyFLpy"
              >
                <button
                  class="sc-jSUZER fgGcDJ"
                  data-testid="hashtag-overview-edit-4"
                >
                  <svg
                    data-testid="edit-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="22.5461"
                      stroke="#013059"
                      stroke-width="3"
                      width="21.0433"
                      x="2.05469"
                      y="7.50781"
                    />
                    <rect
                      fill="white"
                      height="12.0217"
                      width="12.0217"
                      x="15.5781"
                      y="2.99219"
                    />
                    <path
                      d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                      fill="#013059"
                    />
                    <rect
                      fill="#013059"
                      height="2.59198"
                      transform="rotate(-135 29.7031 5.03906)"
                      width="4.53906"
                      x="29.7031"
                      y="5.03906"
                    />
                  </svg>
                </button>
                <button
                  class="sc-jSUZER fgGcDJ"
                  data-testid="hashtag-overview-delete-4"
                >
                  <svg
                    data-testid="delete-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 22 33"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                      stroke="#F23557"
                      stroke-width="3"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="2"
                      x1="0.226562"
                      x2="21.4318"
                      y1="3.72656"
                      y2="3.72656"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="3"
                      x1="7.29688"
                      x2="14.3653"
                      y1="1.82031"
                      y2="1.82031"
                    />
                  </svg>
                </button>
              </div>
            </li>
            <li
              class="sc-eDWCr jtwhuZ"
              data-testid="hashtag-overview-list-item-5"
            >
              <p
                class="sc-pyfCe bsnSHf"
              >
                Wanderlust
              </p>
              <div
                class="sc-bqWxrE kyFLpy"
              >
                <button
                  class="sc-jSUZER fgGcDJ"
                  data-testid="hashtag-overview-edit-5"
                >
                  <svg
                    data-testid="edit-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="22.5461"
                      stroke="#013059"
                      stroke-width="3"
                      width="21.0433"
                      x="2.05469"
                      y="7.50781"
                    />
                    <rect
                      fill="white"
                      height="12.0217"
                      width="12.0217"
                      x="15.5781"
                      y="2.99219"
                    />
                    <path
                      d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                      fill="#013059"
                    />
                    <rect
                      fill="#013059"
                      height="2.59198"
                      transform="rotate(-135 29.7031 5.03906)"
                      width="4.53906"
                      x="29.7031"
                      y="5.03906"
                    />
                  </svg>
                </button>
                <button
                  class="sc-jSUZER fgGcDJ"
                  data-testid="hashtag-overview-delete-5"
                >
                  <svg
                    data-testid="delete-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 22 33"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                      stroke="#F23557"
                      stroke-width="3"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="2"
                      x1="0.226562"
                      x2="21.4318"
                      y1="3.72656"
                      y2="3.72656"
                    />
                    <line
                      stroke="#F23557"
                      stroke-width="3"
                      x1="7.29688"
                      x2="14.3653"
                      y1="1.82031"
                      y2="1.82031"
                    />
                  </svg>
                </button>
              </div>
            </li>
          </ul>
          <div
            class="sc-iAEawV jRdAiF"
          >
            <button
              class="sc-jSUZER fQVznz"
            >
              <svg
                data-testid="arrow-icon"
                fill="none"
                viewBox="0 0 40 26"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                  fill="#013059"
                />
              </svg>
            </button>
            <p
              class="sc-pyfCe bsnSHf"
              data-testid="pagination-counter"
            >
              page1
               / X
            </p>
            <button
              class="sc-jSUZER hjlqfP"
            >
              <svg
                data-testid="arrow-icon"
                fill="none"
                viewBox="0 0 40 26"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                  fill="#013059"
                />
              </svg>
            </button>
          </div>
          <button
            class="sc-hLBbgP kQYyQS"
            data-testid="overview-list-navigate-to-create-page"
          >
            Dodaj hashtag
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-ezOQGI deUqHb"
    >
      <div
        class="sc-cabOPr isyJVs"
      >
        <h2
          class="sc-eDvSVe dcXtK"
        >
          Hashtagi
        </h2>
        <ul
          class="sc-iTFTee bMSplb"
        >
          <li
            class="sc-eDWCr jtwhuZ"
            data-testid="hashtag-overview-list-item-1"
          >
            <p
              class="sc-pyfCe bsnSHf"
            >
              Travel
            </p>
            <div
              class="sc-bqWxrE kyFLpy"
            >
              <button
                class="sc-jSUZER fgGcDJ"
                data-testid="hashtag-overview-edit-1"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-jSUZER fgGcDJ"
                data-testid="hashtag-overview-delete-1"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="sc-eDWCr jtwhuZ"
            data-testid="hashtag-overview-list-item-2"
          >
            <p
              class="sc-pyfCe bsnSHf"
            >
              Adventure
            </p>
            <div
              class="sc-bqWxrE kyFLpy"
            >
              <button
                class="sc-jSUZER fgGcDJ"
                data-testid="hashtag-overview-edit-2"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-jSUZER fgGcDJ"
                data-testid="hashtag-overview-delete-2"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="sc-eDWCr jtwhuZ"
            data-testid="hashtag-overview-list-item-3"
          >
            <p
              class="sc-pyfCe bsnSHf"
            >
              Explore
            </p>
            <div
              class="sc-bqWxrE kyFLpy"
            >
              <button
                class="sc-jSUZER fgGcDJ"
                data-testid="hashtag-overview-edit-3"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-jSUZER fgGcDJ"
                data-testid="hashtag-overview-delete-3"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="sc-eDWCr jtwhuZ"
            data-testid="hashtag-overview-list-item-4"
          >
            <p
              class="sc-pyfCe bsnSHf"
            >
              Nature
            </p>
            <div
              class="sc-bqWxrE kyFLpy"
            >
              <button
                class="sc-jSUZER fgGcDJ"
                data-testid="hashtag-overview-edit-4"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-jSUZER fgGcDJ"
                data-testid="hashtag-overview-delete-4"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="sc-eDWCr jtwhuZ"
            data-testid="hashtag-overview-list-item-5"
          >
            <p
              class="sc-pyfCe bsnSHf"
            >
              Wanderlust
            </p>
            <div
              class="sc-bqWxrE kyFLpy"
            >
              <button
                class="sc-jSUZER fgGcDJ"
                data-testid="hashtag-overview-edit-5"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-jSUZER fgGcDJ"
                data-testid="hashtag-overview-delete-5"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
        </ul>
        <div
          class="sc-iAEawV jRdAiF"
        >
          <button
            class="sc-jSUZER fQVznz"
          >
            <svg
              data-testid="arrow-icon"
              fill="none"
              viewBox="0 0 40 26"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                fill="#013059"
              />
            </svg>
          </button>
          <p
            class="sc-pyfCe bsnSHf"
            data-testid="pagination-counter"
          >
            page1
             / X
          </p>
          <button
            class="sc-jSUZER hjlqfP"
          >
            <svg
              data-testid="arrow-icon"
              fill="none"
              viewBox="0 0 40 26"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                fill="#013059"
              />
            </svg>
          </button>
        </div>
        <button
          class="sc-hLBbgP kQYyQS"
          data-testid="overview-list-navigate-to-create-page"
        >
          Dodaj hashtag
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
